fontdrasil/src/orchestration.rs

```diff
@@ -0,0 +1,102 @@
+//! Common parts of work orchestration.
+
+use std::{
+    collections::HashSet,
+    fmt::Debug,
+    hash::Hash,
+    sync::{Arc, RwLock},
+};
+
+pub const MISSING_DATA: &str = "Missing data, dependency management failed us?";
+
+/// A unit of work safe to run in parallel
+///
+/// Naively you'd think we'd just return FnOnce + Send but that didn't want to compile
+/// See <https://github.com/rust-lang/rust/issues/29625>.
+///
+/// Data produced by work is written into a Context (type parameter C).
+pub trait Work<C, E> {
+    fn exec(&self, context: &C) -> Result<(), E>;
+}
+
+#[derive(Clone)]
+pub struct Cache<T: Default> {
+    pub item: Arc<RwLock<T>>,
```

Comment:
Although I'm not sure exactly how this is used, in general I'd prefer this field not to be `pub`.

Reply:
Deleted Cache